tests/ashley/management/commands/test_sync_group_permissions.py

```diff
@@ -0,0 +1,118 @@
+        )
+
+        with override_settings(
+            ASHLEY_DEFAULT_FORUM_ROLES_PERMISSIONS={
+                "instructor": ["can_see_forum", "can_lock_topics"]
+            }
+        ):
+            # Run the command without argument
+            call_command("sync_group_permissions")
+
+            # By default, the command is in dry-run mode, so it should not have done anything
+            # in database.
+
+            self.assertEqual(
+                ["can_see_forum", "can_read_forum"],
+                self._get_group_forum_permissions(instructor_group, forum),
+            )
+            self.assertEqual(
+                ["can_see_forum", "can_read_forum", "can_approve_posts"],
+                self._get_group_forum_permissions(another_group, forum),
+            )
+
+            # Run the command with the --apply argument, to execute real database updates
+            call_command("sync_group_permissions", "--apply")
+
+            # Check that missing groups have been added to our instructor group, according to
+            # the ASHLEY_DEFAULT_FORUM_ROLES_PERMISSIONS setting
+            self.assertEqual(
+                ["can_see_forum", "can_read_forum", "can_lock_topics"],
+                self._get_group_forum_permissions(instructor_group, forum),
+            )
+            # another_group should have its initial permissions, since it's
+            # not mentionned in the ASHLEY_DEFAULT_FORUM_ROLES_PERMISSIONS setting.
+            self.assertEqual(
+                ["can_see_forum", "can_read_forum", "can_approve_posts"],
+                self._get_group_forum_permissions(another_group, forum),
+            )
+
+            # Run the command with the --remove-extra-permissions to revoke group
+            # permissions that are not defined in the settings
+            call_command(
+                "sync_group_permissions", "--apply", "--remove-extra-permissions"
+            )
+
+            # Check that extra groups have been revoked from instructor group, according to
+            # the ASHLEY_DEFAULT_FORUM_ROLES_PERMISSIONS setting
+            self.assertEqual(
+                ["can_see_forum", "can_lock_topics"],
+                self._get_group_forum_permissions(instructor_group, forum),
+            )
+            # another_group should have its initial permissions, since it's
+            # not mentionned in the ASHLEY_DEFAULT_FORUM_ROLES_PERMISSIONS setting.
+            self.assertEqual(
+                ["can_see_forum", "can_read_forum", "can_approve_posts"],
+                self._get_group_forum_permissions(another_group, forum),
+            )
+
+    @staticmethod
+    def _get_group_forum_permissions(group, forum):
+        return [
+            group_permission.permission.codename
+            for group_permission in GroupForumPermission.objects.filter(
+                forum=forum, group=group, has_perm=True
+            )
+        ]
```

Comment:
There is a method for this in Django's orm :wink: 
```suggestion
        return GroupForumPermission.objects.filter(
            forum=forum, group=group, has_perm=True
        ).values_list("permission__codename", flat=True)
```

Reply:
Thanks for the tip ! :pray: